lib/posix-process/process.c

```diff
@@ -281,19 +286,30 @@ int tcsetpgrp(int fd __unused, pid_t pgrp)
 	}
 	return 0;
 }
+#endif /* UK_LIBC_SYSCALLS */
 
+#if UK_LIBC_SYSCALLS
 pid_t tcgetpgrp(int fd)
 {
 	/* We have a single "process group" */
 	return UNIKRAFT_PGID;
 }
+#endif /* UK_LIBC_SYSCALLS */
+
+UK_SYSCALL_R_DEFINE(int, nice, int, inc)
+{
+	/* We don't support priority updates for unikernels */
+	return -EPERM;
+}
 
+#if UK_LIBC_SYSCALLS
 int nice(int inc __unused)
 {
 	/* We don't support priority updates for unikernels */
-	errno = EPERM;
+	errno = uk_syscall_r_nice(inc);
```

Comment:
Hum, this call is making the errno value negative. Secondly, `UK_SYSCALL_R_DEFINE(int, nice, int, inc)` is already provide `nice` so this wrapper should be removed completely.

Reply:
I agree on the `nice` syscall, but the other functions are not syscalls. For example `wait3` is defined in musl as follows:
```
pid_t wait3(int *status, int options, struct rusage *usage)
{
        return wait4(-1, status, options, usage);
}
```

Reply:
Fair enough, this makes sense.